cnf-certification-test/platform/suite.go

```diff
@@ -292,21 +296,26 @@ func testTainted(env *provider.TestEnvironment) {
 		if err != nil {
 			tnf.ClaimFilePrintf("failed to get tainter modules from node %s: %v", nodeName, err)
 			errNodes = append(errNodes, nodeName)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewNodeReportObject(nodeName, "Failed to get tainter modules", false))
 			continue
 		}
 
 		// Save modules' names only.
 		for moduleName, taintsLetters := range tainters {
 			moduleTaints := nodetainted.DecodeKernelTaintsFromLetters(taintsLetters)
 			badModules[nodeName] = append(badModules[nodeName], badModuleTaints{name: moduleName, taints: moduleTaints})
-
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewNodeReportObject(nodeName, "Module taints kernel", false).
+				AddField(testhelper.ModuleName, moduleName).
+				AddField(testhelper.Taints, strings.Join(moduleTaints, ",")))
 			tnf.ClaimFilePrintf("Node %s - module %s taints kernel: %s", nodeName, moduleName, moduleTaints)
 		}
 
 		// Lastly, check that all kernel taint bits come from modules.
 		otherKernelTaints := nodetainted.GetOtherTaintedBits(taintsMask, taintBitsByAllModules)
 		for _, taintedBit := range otherKernelTaints {
 			tnf.ClaimFilePrintf("Node %s - taint bit %d is set but it's not caused by any module.", nodeName, taintedBit)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewNodeReportObject(nodeName, "Taint bit is set but it's not caused by any module", false).
+				AddField(testhelper.TaintBit, strconv.Itoa(taintedBit)))
```

Comment:
You could also add another field with the taint bit description, using package "nodetainted.getTaintMsg(int bit)" function, changing first the "g" to "G" to make it a package's public func.